### DISCUSSION TITLE
Social login/Registration using OCAPI

### DISCUSSION THREAD

**Discussion by amitkathpal123** [SalesforceCommerceCloud/sfcc-ci#521] [Team Posts]
Can someone please direct me to step by step approach or sample code in order to implement social login and registration using OCAPI.

**Comment by bmoraux** [comment]
Hello,

Social Login cannot be implement trough OCAPI be cause external OAUTH authentification is not support by OCAPI right now. 

To register customer trough OCAPI, you need : 
- Auth as a Guest (POST /customers/auth)
- Create an account (POST /customers)

Regards

**Comment by amitkathpal123** [comment]
Thank you for responding, @bmoraux.

In documentation (link below), "/customers", it reads you can register a customer and in case of OAUTH, don't provide password. Doesn't it mean OCAPI supports OAUTH or if it meant something else. Can you please help clarifying.

https://documentation.b2c.commercecloud.salesforce.com/DOC1/index.jsp?topic=%2Fcom.demandware.dochelp%2FOCAPI%2F19.1%2Fshop%2FResources%2FCustomers.html&cp=0_12_3_5

**Comment by bmoraux** [comment]
Hello @amitkathpal123,

You need to understand the difference between JWT and OAUTH, 
https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/OCAPI/19.1/usage/JWT.html
https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/OCAPI/19.1/usage/OAuth.html

- JWT is to generate token for customer  
- OAUTH is for BM and server to server authentification

Regarding /customers, when you are in customer context you need a JWT guest and this particular case required password
When you are in BM context (with OAUTH context), you can create customer without password (but as an admin)

Regards

**Comment by BrillioAnshuman** [comment]
@amitkathpal123 ,
@bmoraux 
were you able to figure out a solution for this? I too have similar kind of requirement to have user login using several social media account using oAuth.

detailed steps are much appreciated.
